### PR TITLE
CXF-8618: Update to Spring Boot 2.6.0

### PR DESCRIPTION
**Pull request by reta**
Update to Spring Boot 2.6.0:
   - Update to Spring Boot 2.6.0 (https://spring.io/blog/2021/11/19/spring-boot-2-6-is-now-available)
   - Update to Spring Security 5.6.0 (https://spring.io/blog/2021/11/16/spring-security-5-6-0-released)
   - Update to Micrometer 1.8.0 (https://github.com/micrometer-metrics/micrometer/releases/tag/v1.8.0)

### CODE REVIEW

File: distribution/src/main/release/samples/pom.xml [hunk 1]
```diff
@@ -30,8 +30,8 @@
         <!-- don't deploy the samples, kind of pointless -->
         <maven.deploy.skip>true</maven.deploy.skip>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <spring.boot.version>2.5.7</spring.boot.version>
-        <spring.cloud.eureka.version>3.0.3</spring.cloud.eureka.version>
+        <spring.boot.version>2.6.0</spring.boot.version>
+        <spring.cloud.eureka.version>3.1.0-M3</spring.cloud.eureka.version>
```
Comment (reta) [inline distribution/src/main/release/samples/pom.xml:34]: Release is expected at the end of November

**Comment by reta** [comment]
@coheigea @amarkevich any concerns, guys? Upgrades cleanly, no issues (3.5.0 only), thanks!

**Review by amarkevich — APPROVED** [review]
LGTM

**Review by deki — APPROVED** [review]
Thanks @reta

**Pull request merged**